server/game_thread.cpp

```diff
@@ -462,13 +482,35 @@ void gameOnMessage(dfws::Handle hdl, const std::string& payload)
 		executeCommandKill(*p, event->mobID());
 	}
 	break;
+	case FlatBuffGenerated::ClientMessageUnion::ClientMessageUnion_CommandSkill:
+	{
+		const auto event = clientMessage->event_as_CommandSkill();
+		executeSkill(*p, event->skill(), {event->mousePos()->x(), event->mousePos()->y()});
+	}
+	break;
 
 	default:
 		std::cout << "gameOnMessage: some other message type received\n";
 		break;
 	}
 }
 
+template<typename C>
+void updateCollideables(std::vector<std::unique_ptr<C>>& collideables) {
+	std::vector<int> despawns;
+	for (size_t i = 0; i < collideables.size(); i++)
+	{
+		collideables[i]->update();
+		if (collideables[i]->toBeDeleted)
+			despawns.push_back(i);
+	}
+	for (int i = despawns.size() - 1; i >= 0; i--)
+	{
+		std::cout << "erasing collideable\n";
+		collideables.erase(collideables.begin() + despawns[i]);
+	}
```

Comment:
Use `std::erase_if`:

```c++
std::erase_if(collideables.begin(), collideables.end(), [] (auto& c) {
    c->update();
    return c->toBeDeleted;
});
```

Reply:
```
/deadfish/server/game_thread.cpp:506:7: error: no member named 'erase_if' in namespace 'std'
        std::erase_if(collideables.begin(), collideables.end(), [] (auto& c) {
        ~~~~~^
```
I think we have to upgrade to c++20 as soon as possible

Reply:
```
	for (auto it = collideables.begin(); it != collideables.end();) {
		(*it)->update();
		if ((*it)->toBeDeleted) {
			collideables.erase(it);
		} else {
			++it;
		}
	}
```
seems to be working here as well (version with `for(...; ...; ++it)` and `collideables.erase(it); --it;` too), but probably because the changes during testing weren't critical enough to cause vector reallocation. Can somebody (@piodul, @capturcus) verify my thinking here? :D